### ISSUE TITLE
Intellisence library include issues

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode-cpptools#764]
_From @Cesaar on May 23, 2017 23:52_

- VSCode Version: Code 1.12.2 (19222cdc84ce72202478ba1cec5cb557b71163de, 2017-05-10T13:20:36.315Z)
- OS Version: Windows_NT ia32 10.0.15063
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|code-gnu-global|austin|0.2.2|
|vscode-clang|mitaki28|0.2.2|
|cpptools|ms-vscode|0.11.1|;

---

Steps to Reproduce:

1. Create a new project
2. include any of the standard libraries.
![capture](https://cloud.githubusercontent.com/assets/7192563/26381220/76351f52-4023-11e7-898b-206696f842f4.PNG)

As you can see, the editor won't recognise the include path. However, the project builds fine.
This started happening after updating to the C/C++ extension version 0.11.1

_Copied from original issue: Microsoft/vscode#27168_

**Comment by bobbrow** [comment]
@Cesaar, can you share your includePath?

**Comment by sjiekak** [comment]
I have a similar issue. After update to 0.11.1 intellisence keep signaling errors while there are none.

**Comment by Cesaar** [comment]
@bobbrow 
sure:
[c_cpp_properties.txt](https://github.com/Microsoft/vscode-cpptools/files/1024935/c_cpp_properties.txt)

This is the automatically generated configurations file however, I didn't change anything in it. And before the update it was working fine.

On my laptop I'm using Linux. I opened the same project before updating, everything fine. But then after updating to 0.11.1 I see the same problems. So as far as I know it's not platform specific.

Thanks in advance.

**Comment by portwaypoint** [comment]
Same here since upgrading to 0.11.1

This is on a Linux host.

It's almost as if it is permanently hard-coded to search "/usr/include" on Linux since that appears to be the only place it will find header files successfully.

```
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "src/include",
                "/usr/include"
            ],
            "browse": {
                "limitSymbolsToIncludedHeaders": false,
                "databaseFilename": "",
                "path": [
                    "src/include",
                    "/usr/include"
                ]
            }
        }
    ]
}
```
It won't find headers in "src/include" for instance (workspace relative)

**Comment by sean-mcmanus** [comment]
@portwaypoint You can use syntax like `"${workspaceRoot}/src/include"` for workspace relative paths. You may want to create a new c_cpp_properties.json to see the default paths we supply.

**Comment by bobbrow** [comment]
@Cesaar, your includePath looks incomplete.  If you delete your c_cpp_properties.json file and generate a new one (In the command palette, select "C/CPP: Edit Configurations"), it should pick up the right files.

**Comment by portwaypoint** [comment]
@sean-mcmanus Adding ${workspaceRoot} seems to have done the trick for me, thanks for pointing that out. Still a little confused as to why it only started to fail to pickup the includes after upgrading to 0.11.1 (co-incidence perhaps).

However, extra vote for downgrade option, as at least that would allow us to eliminate whether it is the plugin or user error.

**Comment by Cesaar** [comment]
@bobbrow , I regenerated the file, but it is identical to the one I had.

Meanwhile, I've updated both versions ov Visual Studio Code, Linux and Windows, with the 0.11.2 patch.
On Linux, everything is fine now.

On windows I'm still seeing the issue.
When you say my includePath looks incomplete, what is missing hten?

Andreas

**Comment by bobbrow** [comment]
Your include path does not include a path to any of the STL headers.  Usually this comes from Visual Studio when you are on Windows, e.g:
`C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include/*`
or
`C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Tools/MSVC/14.10.25017/include/*`

Do you have any version of Visual Studio installed?  The extension currently detects the CRT path from VS 2015 and VS 2017 if they are installed.  If you are using an older version of Visual Studio, then you’ll need to add the appropriate include path (it will look similar to the VS 2015 path)

**Comment by Cesaar** [comment]
OK, now it's all working fine.
I did have an installation of Visual Studio 2012, so that fixed my problem.

Thanks for all the help!

**Comment by bobbrow** [comment]
Glad to hear that worked for you.

@portwaypoint, there is no downgrade option, but we [included a way to revert back to the previous behavior](https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/LanguageServer/FAQ.md#why-do-i-have-red-underlines-everywhere-after-updating-to-the-latest-version) if you are unsatisfied with the new IntelliSense.

In any case, since it seems many users were unable to find this setting on their own, we have temporarily reverted the default behavior back to Tag Parser for everyone with 0.11.2 until we can improve the migration experence.  For those who have fixed up their include paths to work with 0.11.1, they can turn the new IntelliSense back on by adding `"C_Cpp.intelliSenseEngine": "Default"` to their settings.json file.
